### ISSUE TITLE
wysihtml5 doesn't support ie6-7

### ISSUE THREAD

**Issue by nicoburns** [tiff/wysihtml5#147]
The title says it all really. Is there any reason for not supporting 6 and 7? It shouldn't be much extra work as you are already supporting 8, especially as you are using the Rangy library.

I don't want to presume that there aren't any issues, because I know these browsers can be a pain, but I've tended to find that contenteditable wise, there's nothing that 8 supports that doesn't also work in 6 and 7. I think it would help to make the editor more widely accepted if this support could be achieved.

**Comment by tiff** [comment]
I was just waiting for this issue to be created :)
What I can definitely say is that I will never (again) put any effort in supporting IE < 8.
When I started wysihtml5 I was trying to support at least IE7. It ended up horrible.
You're right: There not so many differences in the implementation of contentEditable and the selection api between IE8 and IE < 8.
What I know is that IE < 8 randomly crashes in contentEditable mode. Also it doesn't support the selector api (`document.querySelector`) which wysihtml5 highly benefits from.
Also if you wanna support those browsers you will have to deal with a lot more invalid markup (`<font>`, `<center>`, ...).

To sum it up:
wysihtml5 should stay lightweight. Supporting legacy browsers will make it more clunky.
I don't know if you are aware of it but IE < 8 users will still see a `<textarea>` (Graceful degradation)

If your company forces you that everything in IE6+7 should work and look the same as in modern browsers then you might wanna use a different library or work on a fork of wysihtml5.
Sorry :v:

BTW: My plan for the future is to even fade out support for IE 8 in order to get rid of rangy.

**Comment by nicoburns** [comment]
Ha, fair enough :)
What you say certainly makes sense, especially if you are planning to remove rangy too.

I have actually made my own WYSIWYG editor - http://ghosted.it/demo (also MIT licensed, so obviously feel free to copy any code) - so I will probably continue to use that, but wysihtml5 is definitely an option for projects which don't require legacy browser support.

Anyway, it was worth asking I guess :)
